content/toolsAndResources/nge.md

```diff
@@ -0,0 +1,130 @@
+
+Input is just one type of several other node types such as contextual, logical, math, and more. Connecting these different node types to the source nodes can produce an almost limitless variety of meshes. 
+
+## Adding Nodes
+
+Adding nodes to the graph is one of the most common actions taken while working with the editor. There are two main methods for adding node to the graph. The first method is clicking and dragging a node from the node list onto the graph and releasing the mouse button. This will drop the selected node at the mouse position. The node list can be filtered by entering a keyword in the search bar at the top of the list to help find a node faster.
+
+The second method is to press the spacebar for a shortcut to open a search window with a the full node list at the mouse position on the graph. Once the window is open, we can simply start typing to enter a keyword in the search bar. Once the list is filtered, use either the mouse to click the desired node or use the arrow keys to highlight the desired node and press enter. This will drop the node at the mouse position when the shortcut was invoked. If a wire was selected when the shortcut was opened, the editor will attempt to connect the node in the middle of the connection that was selected. If the node to be placed contains compatible inputs and outputs, a connection will be made.
+
+![Press spacebar for shortcut to search bar to filter node list which appears at the mouse position](/img/tools/nge/nodeAddShortcut.jpg)
+
+There is one other shortcut to quickly add nodes in a contextual manner. For most node inputs with a defined type, we can simply click on the input and drag to the left of the node which will drag out a dashed wire of the type the input expects. Once the mouse button is released, an input node of the correct type will be dropped. For example, the image below illustrates clicking on the height input of the `Box` source node and dragging a dashed line to the left of the node. Once the mouse button is released, a `Float` node is added to the graph at the mouse position.
+
+![Click and drag from node input to drop an automatically connected node of the appropriate type on release of the mouse button](/img/tools/nge/dragAddNode.jpg)
+
+## Connecting Nodes Together
+All nodes have one or more inputs and one or more outputs, however each individual node has requirements for the data that passes through it. These requirements typically take the form of data types that the nodes expect for their calculation. To easily tell what type of data a node expects is to look at the inputs and output icons on the node. To help decipher the icons, please refer to the legend below:

```

Comment:
"All nodes have one or more inputs and one or more outputs": not totally exact, as input nodes don't have inputs and the "Geometry output" node does not have any output. I don't think it's worth a change, though.

Reply:
Rewrote the sentence and broke it into two sentences to address input and output nodes without too much depth in explanation.